site/source/_posts/2019-12-09-web3-what-are-your-options.md

```diff
@@ -0,0 +1,194 @@
+
+***(Yes, I realise that rhymes.  No, I didn't realise until my second proof-read through of this article!)***
+
+If we were going deeper into Ethereum Smart Contract development, or doing anything more advanced than this intro post, I would be using [Embark](https://embark.status.im/), but then potentially Ganache alongside it.
+
+Anyway, to install Ganache head over to [this page](https://www.trufflesuite.com/ganache) and click on the executable there.  If you so choose; there is also a Ganache CLI available you can install by running:
+
+```
+npm install -g ganache-cli
+```
+
+Running the Ganache CLI will give you the same functionality as the desktop client; in essence giving us a multitude of ETH-loaded wallets that we can build contracts around / interact with.
+
+![Ganache CLI](../assets/images/ganache-cli.png)
+
+Rather brilliantly; we now have a local Ethereum Node running that we can start using the Web3 client to interact with.  In another Terminal tab, open up a `node` instance from the same working directory we ran the `npm init` command from earlier.
+
+Now, in our interactive Node console, run:
+
+``` js
+var Web3 = require('web3');
+var web3 = new Web3('http://localhost:8545');
+```
+
+Something to note here, is that I'm calling `new Web3` with an `http` protocol, but the WebSocket protocol is also commonly used:
+
+``` js
+var web3 = new Web3(Web3.givenProvider || new Web3.providers.WebsocketProvider('ws://remotenode.com:8546'));
+``` 
+
+To test and ensure the connection, you can get a list of the wallets made available in Ganache by running:
+
+``` js
+web3.eth.getAccounts().then(console.log);
+```
+
+Which *should* give you an output like the following:
+
+```js
+> [ '0x7689cF9F90FAad61B8a3c9b1b2A5b4580B37358b',
+  '0x852e9a9db77a4e6169e175cDBb33dBE350150A8e',
+  '0x946700a1a4f30Dfe80307C91B6DaF1cCa2d09401',
+  '0x7d356aF02A87147D3ce5F9ACA716a78f70aF7616',
+  '0x88A116a16e4c8838F575a0e7a72eE27C7B073263',
+  '0x655317701Fcf3b310F492cB801C8D23f8c6fb556',
+  '0x16D305e72aFb0DDa1dB1830F8a98D5cD5337882E',
+  '0x9099bb4Af9CE5734E7a7e62f817e833fcFFaaF32',
+  '0x2ec4CC6700d0424A78a9B9Fc2ecBaeFc162313F1',
+  '0x1BC51a0edEC9FdEA3B14748e9209F4bF8Fe024b5' ]
+```
+
+If you want to check the balance of an individual wallet from the above list, you can do so by running:
+
+```js
+const account1 = "0x7689cF9F90FAad61B8a3c9b1b2A5b4580B37358b";
+
+web3.eth.getBalance(account1)
+.then(console.log);
+```
+
+Which will output:
+
+```
+> 100000000000000000000
+```
+
+
+## Contract Interaction
+
+As above; interacting with our *individual wallets* through `web3.js` is cool, but not nearly the extent to which the library works.  Let's now take a brief look at the more important functionality; of interacting with smart contracts through `web3.js`.
```

Comment:
"wallets" -> "accounts"

Also, so far, at least in the documentation, we've used "Smart Contracts" as a name (not "contracts" as too ambiguous and not "smart contracts") maybe we can use the chance to update this as well?